### DISCUSSION TITLE
Catch JSON and CSS warnings

### DISCUSSION THREAD

**Discussion by jrock2004** [mswjs/msw#1310] [Q&A]
So I see things like then when I make a code change and my page refreshes with change. What is the best way to catch these things?

![image](https://user-images.githubusercontent.com/655716/177021214-27e2deed-8e56-4115-b190-7cba10f44515.png)

**Comment by kettanaito** [comment]
Hey, @jrock2004. Thanks for raising this.

It's hard for MSW to assume anything about the requests it intercepts. There may be requests to `*.json` that the user does want to catch and mock, so we cannot rule out all JSON or CSS or any other static requests. 

What you can do is to rule them out on your side, because you're closer to the request source:

```js
worker.start({
  onUnhandledRequest(request, print) {
    // Do not treat requests to JSON files as unhandled.
    if (request.url.path.endsWith('.json')) {
      return
    }

    print.warning()
  }
})
```

I know this may feel verbose but it's much safer than assuming this as a part of the library's logic. 

I would love to ship a better default experience but this kind of assumption is always hard. Things like `*.hot-update.json` is an implementation detail and webpack (or any other dev server) may change that without notice, breaking the logic we build around it. If you know a reliable way to bypass certain static assets please share it in this thread.

**Reply by jrock2004** [reply]
So I tried doing the following and getting errors that request.url does not exist

```
const prepare = async (): Promise<void> => {
  if (process.env.NODE_ENV === 'development') {
    const { worker } = await import('./mocks/browser');

    worker.start({
      onUnhandledRequest: (request: any, print: any): void => {
        if (request && request.url.path.startsWith('https://cloudflare-ipfs.com/ipfs')) {
          return;
        }

        print.warning();
      },
    });
  }
};
```

**Reply by kettanaito** [reply]
@jrock2004 debug what `request` value is. It's always a request representation given to you by MSW. I'd rather say that `request.url.path` should be `request.url.pathname`.
